.github/workflows/rust.yml

```diff
@@ -178,3 +202,32 @@ jobs:
       - name: Mark the job as unsuccessful
         if: ${{ contains(needs.*.result, 'failure') || contains(needs.*.result, 'cancelled') }}
         run: exit 1
+
+  publish-release:
+    name: Publish release
+    runs-on: ubuntu-latest
+    needs:
+      ["linux", "mac", "windows"]
+    if: ${{ !contains(needs.*.result, 'failure') && !contains(needs.*.result, 'cancelled') }}
```

Comment:
This condition is not needed here, as this job is not run if any needed fails. This whole job should have if guard from step bellow.

Reply:
I can remove this if guard. But I think if I move the if guard from steps below, verify job might be [skipped](https://github.com/wusyong/mozjs/actions/runs/8135726635) as well if this release job is skipped.

Reply:
I think skipped is mapped to cancelled result.